src/main/java/uk/gov/hmcts/reform/hmc/api/model/ccd/BaseCaseData.java

```diff
@@ -0,0 +1,29 @@
+package uk.gov.hmcts.reform.hmc.api.model.ccd;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.experimental.SuperBuilder;
+
+import java.time.LocalDateTime;
+
+@Data
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@Getter
+@AllArgsConstructor
+@NoArgsConstructor
+@Setter
+@SuperBuilder(toBuilder = true)
+public class BaseCaseData {
```

Comment:
is this class needed?

Reply:
I need Id field to get case reference number, for that i used this basecase data to get caseid for populating case ref but now am removing this basecasedata class and using id directly in case data .. Thank you @yogendraU